pkg/resource/grafana-agent-config/reconciler.go

```diff
@@ -0,0 +1,95 @@
+package grafanaagentconfig
+
+import (
+	"context"
+	"reflect"
+
+	"github.com/pkg/errors"
+	v1 "k8s.io/api/core/v1"
+	apimachineryerrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+
+	loggedcluster "github.com/giantswarm/logging-operator/pkg/logged-cluster"
+	promtailclient "github.com/giantswarm/logging-operator/pkg/resource/promtail-client"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+// Reconciler implements a reconciler.Interface to handle
+// GrafanaAgent config: extra grafana-agent config defining what we want to retrieve.
+type Reconciler struct {
+	client.Client
+}
+
+// ReconcileCreate ensures grafana-agent config is created with the right credentials
+func (r *Reconciler) ReconcileCreate(ctx context.Context, lc loggedcluster.Interface) (ctrl.Result, error) {
+	logger := log.FromContext(ctx)
+	logger.Info("grafana-agent-config create")
+
+	// Retrieve promtail credentials
+	var promtailSecret v1.Secret
+	err := r.Client.Get(ctx, types.NamespacedName{Name: promtailclient.SecretMeta(lc).Name, Namespace: promtailclient.SecretMeta(lc).Namespace},
+		&promtailSecret)
+	if err != nil {
+		return ctrl.Result{}, errors.WithStack(err)
+	}
+
+	promtailCredentials, err := promtailclient.GetPromtailCredentials(lc, &promtailSecret)
```

Comment:
Maybe we should rename promtailCredentials to loki-client-credentials or logging-agent-credentials?

Reply:
I think we should check all things that are named promtail even :)